testgrid/util/gcs/read.go

```diff
@@ -0,0 +1,216 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gcs
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"regexp"
+	"sync"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/iterator"
+
+	"k8s.io/test-infra/testgrid/metadata"
+	"k8s.io/test-infra/testgrid/metadata/junit"
+)
+
+// Started holds started.json data.
+type Started = metadata.Started
+
+// Finished holds finished.json data.
+type Finished struct {
+	metadata.Finished
+	Running bool
```

Comment:
:thinking: why we have a finished.json while the job is still running?

Reply:
I'll clean this up later, this is an internal thing to help determine if the job is finished (rather than looking at the finished timestamp